### PR TITLE
Fix Bugs

### PR DESCRIPTION
**Pull request by illuzen**
Many tests were failing in the chain when we connected this to it. It was mostly due to PrefixedMemoryDB and MemoryDB having .default() return a single null byte instead of 8 null bytes. So we overwrote the Default trait by wrapping these types. We also remove a sloppy patch that was trying to address this from before. There were also a bunch of corner cases that were not tested in this repo but were in sp-state-machine so we added similar tests here.

### CODE REVIEW

**Comment by illuzen** [comment]
Also, we just make a compact proof wrap a storage proof instead of trying to make a felt-aligned compact format. This can be addressed later if necessary.

File: src/storage_proof.rs [hunk 1]
```diff
@@ -256,4 +260,13 @@ pub mod tests {
...
         ));
     }
+
+    #[test]
+    fn invalid_compact_proof_does_not_panic_when_decoding() {
+        let invalid_proof = CompactProof {
+            encoded_nodes: vec![vec![135]],
+        };
+        let result = invalid_proof.to_memory_db::<Hasher>(None);
+        assert!(result.is_ok()); // Should not panic, even with invalid data
```
Comment (n13) [inline src/storage_proof.rs:270]: Why would an invalid proof not panic? 

Shouldn't it panic??

Comment (illuzen) [reply src/storage_proof.rs:270]: We don't do any checking of proof validity on the conversion to and from a compact proof anymore.
